### PR TITLE
CANS-725 - Fix eslint warnings

### PR DESCRIPTION
**Pull request by wmitchell-cambria**
## JIRA

[CANS-725 - Fix eslint warnings](https://osi-cwds.atlassian.net/browse/CANS-725)

## Description

This resolves all lint warnings in the CANS project at time of authoring. The lints were resolved in the following ways:
1. Adding the ability to lint `*.test.js` files differently, since certain things that are bad practice in source files are perfectly acceptable in unit tests. (Primarily, magic numbers in test files are ignored.)
2. Manually fixing magic numbers in source files.
3. Refactoring SubordinateLoadingBoundary to use the common LoadingBoundary base. It was originally authored before the base LoadingBoundary existed, and it has a couple (technically irrelevant) lint warnings.

## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests
The only functional change I made was to the SubordinateLoadingBoundary, which has a corresponding unit test.

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/Assessment/ItemRegularRating.js [hunk 1]
```diff
@@ -16,12 +16,13 @@ const PrimRadio = withStyles({
   disabled: {},
 })(Radio)
 
-const regularRating = 4
+const numRegularRatings = 4
+const naRatingValue = 8
```
Comment (AK56) [inline app/javascript/Application/components/Assessment/ItemRegularRating.js]: We are using this constant in multiple files, perhaps we should declare it in a helper

Comment (wmitchell-cambria) [reply app/javascript/Application/components/Assessment/ItemRegularRating.js]: I was thinking the same thing, but it felt silly to create a helper just for that. I'm creating a followup PR to refactor these ItemRegularRating and ItemBooleanRating components to share more code. I can add (or create another PR) to clean up the N/A rating as well.

**Comment by wmitchell-cambria** [comment]
I will do a second pass on the variable naming.

File: app/javascript/Application/components/Assessment/ItemRegularRating.js (hunk 1 above)
Comment (AK56) [reply app/javascript/Application/components/Assessment/ItemRegularRating.js]: And it gets better :-)

**Pull request merged**